nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/main/java/org/nuxeo/ecm/core/storage/sql/RowMapper.java

```diff
@@ -342,20 +343,23 @@ CopyResult copy(IdWithTypes source, Serializable destParentId, String destName,
 
         public final Serializable targetId;
 
-        public final boolean isRetentionActive;
+        public final boolean isUndeletable;
 
         /**
          * Creates node info for a node that may also be a proxy.
          */
         public NodeInfo(Serializable id, Serializable parentId, String primaryType, Boolean isProperty,
-                Serializable versionSeriesId, Serializable targetId, boolean isRetentionActive) {
+                Serializable versionSeriesId, Serializable targetId, Calendar retainUntil, boolean hasLegalHold,
+                boolean isRetentionActive) {
             this.id = id;
             this.parentId = parentId;
             this.primaryType = primaryType;
             this.isProperty = isProperty;
             this.versionSeriesId = versionSeriesId;
             this.targetId = targetId;
-            this.isRetentionActive = isRetentionActive;
+            isUndeletable = hasLegalHold //
+                    || (retainUntil != null && Calendar.getInstance().before(retainUntil)) //
```

Comment:
we can related only on `Calendar.getInstance().before(retainUntil))` as it null safe: 
```
public boolean before(Object when) {
        return when instanceof Calendar
            && compareTo((Calendar)when) < 0;
    }
```

Reply:
`retainUntil != null` is here on purpose because it's a short-circuit check much faster than instantiating a `Calendar` instance.

Reply:
`retainUntil != null` is here on purpose because it's a short-circuit check much faster than instantiating a `Calendar` instance.